packages/taco-auth/src/providers/eip4361/external-eip4361.ts

```diff
@@ -1,18 +1,33 @@
 import { SiweMessage } from 'siwe';
 
-import { EIP4361_AUTH_METHOD, EIP4361AuthSignature } from './auth';
+import { AuthProvider } from '../../auth-provider';
 
-export const USER_ADDRESS_PARAM_EXTERNAL_EIP4361 =
-  ':userAddressExternalEIP4361';
+import { EIP4361_AUTH_METHOD, EIP4361AuthSignature } from './auth';
 
-export class SingleSignOnEIP4361AuthProvider {
+/**
+ * SingleSignOnEIP4361AuthProvider handles Sign-In with Ethereum (EIP-4361/SIWE) authentication
+ * using an existing SIWE message and signature.
+ *
+ * This provider validates and reuses an existing SIWE message and signature rather than generating new ones.
+ * It's useful for implementing single sign-on flows where the SIWE authentication was performed elsewhere.
+ */
+export class SingleSignOnEIP4361AuthProvider implements AuthProvider {
+  /**
+   * Creates a new SingleSignOnEIP4361AuthProvider from an existing SIWE message and signature.
+   *
+   * @param existingSiweMessage - The existing SIWE message string to validate and reuse
+   * @param signature - The signature corresponding to the SIWE message
+   * @returns A new SingleSignOnEIP4361AuthProvider instance
+   * @throws {Error} If signature verification fails or message parameters are invalid
+   */
   public static async fromExistingSiweInfo(
     existingSiweMessage: string,
     signature: string,
   ): Promise<SingleSignOnEIP4361AuthProvider> {
-    // validation
     const siweMessage = new SiweMessage(existingSiweMessage);
+    // this will trigger validation for the signature and all parameters (`expirationTime`, `notBefore`...).
     await siweMessage.verify({ signature });
```

Comment:
Should we be checking here for 2-hour expiry? We didn't before and I believe you had a suggestion to do so, which I think is a good one. I believe you can include the two-hour time check by passing a 2-hour time as one of the `VerifyParams` - https://github.com/spruceid/siwe/blob/main/packages/siwe/lib/client.ts#L270.

Also needs to be done in `getOrCreateAuthSignature()`.

Reply:
If you add it, also add a test to ensures it works properly.